pdata/README.md

```diff
@@ -0,0 +1,214 @@
+# Pipeline data (pdata)
+
+Pipeline data (pdata) implements data structures that represent telemetry data in-memory. All data received 
+is converted into this format, travels through the pipeline in this format, and is converted from this format by 
+exporters when sending.
+
+Current implementation primarily uses OTLP protobuf structs as the underlying data structures for many of the 
+declared structs. We keep a pointer to OTLP protobuf in the "orig" member field. This allows efficient translation 
+to/from OTLP wire protocol. The underlying data structure is kept private so that we are free to make changes to it 
+in the future.
+
+API provided by pdata ensures that no underlying data can be used in more than one place. Each pdata instance cannot 
+contain a reference to an object that is already used somewhere else.
+
+## API naming convention
+
+### Protobuf message representation in pdata
+
+Packages of pdata are not following names of the protobuf packages. pdata has a package per telemetry type starting 
+with `p`, e.g. `ptace`, and `pcommon` package with API for other protobuf definitions from `common` and `resource` 
+protobuf packages.
+
+Pipeline data structs SHOULD be based on the names of the underlying OTLP protobuf messages. Data types for 
+protobuf messages defined as part of another message SHOULD include the owner's name as a prefix. Exceptions to these 
+rules are possible, but not encouraged. Another name can be chosen for brevity or if the struct provides a completely
+different interface to manipulate the underlying data. The following exceptions are currently accepted:
+
+- `plog.Logs` based on `LogsData` protobuf message.
+- `pmetric.Metrics` based on `MetricsData` protobuf message.
+- `ptrace.Traces` based on `TracesData` protobuf message.
+- `pcommon.Slice` based on `ArrayValue` protobuf message.
+- `pcommon.Map` based on `KeyValueList` protobuf message.
+- `pcommon.Value` based on `AnyValue` protobuf message.
```

Comment:
Would it be better to delete all these lines and only keep one exception example, to reduce the amount of maintenance this document requires if the source code is refactored? Let the source code be the source of truth. If you want the fact of the exception to be documented it would be better to add a comment in the source code near the name.

Reply:
Makes sense. Removed